src/particle.cpp

```diff
@@ -0,0 +1,681 @@
+#include "openmc/particle.h"
+
+#include <algorithm> // copy, min
+#include <cmath>     // log, abs, copysign
+#include <sstream>
+
+#include "openmc/bank.h"
+#include "openmc/capi.h"
+#include "openmc/cell.h"
+#include "openmc/constants.h"
+#include "openmc/error.h"
+#include "openmc/geometry.h"
+#include "openmc/hdf5_interface.h"
+#include "openmc/material.h"
+#include "openmc/message_passing.h"
+#include "openmc/mgxs_interface.h"
+#include "openmc/nuclide.h"
+#include "openmc/photon.h"
+#include "openmc/physics.h"
+#include "openmc/physics_mg.h"
+#include "openmc/random_lcg.h"
+#include "openmc/settings.h"
+#include "openmc/surface.h"
+#include "openmc/simulation.h"
+#include "openmc/tallies/derivative.h"
+#include "openmc/tallies/tally.h"
+#include "openmc/tallies/tally_scoring.h"
+#include "openmc/track_output.h"
+
+namespace openmc {
+
+//==============================================================================
+// LocalCoord implementation
+//==============================================================================
+
+void
+LocalCoord::rotate(const std::vector<double>& rotation)
+{
+  this->r = this->r.rotate(rotation);
+  this->u = this->u.rotate(rotation);
+  this->rotated = true;
+}
+
+void
+LocalCoord::reset()
+{
+  cell = C_NONE;
+  universe = C_NONE;
+  lattice = C_NONE;
+  lattice_x = 0;
+  lattice_y = 0;
+  lattice_z = 0;
+  rotated = false;
+}
+
+//==============================================================================
+// Particle implementation
+//==============================================================================
+
+Particle::Particle()
+{
+  // Create and clear coordinate levels
+  coord_.resize(model::n_coord_levels);
+  cell_last_.resize(model::n_coord_levels);
+  clear();
+
+  for (int& n : n_delayed_bank_) {
+    n = 0;
+  }
+
+  // Create microscopic cross section caches
+  neutron_xs_.resize(data::nuclides.size());
+  photon_xs_.resize(data::elements.size());
+}
+
+void
+Particle::clear()
+{
+  // reset any coordinate levels
```

Comment:
Follow consistency in comments, either a sentence starts with an upper case everywhere or lower case everywhere.